src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncIteratorRewriter.cs

```diff
@@ -139,7 +139,7 @@ protected override void GenerateControlFields()
                 // Add a field: bool disposeMode
                 _disposeModeField = F.StateMachineField(boolType, GeneratedNames.MakeDisposeModeFieldName());
 
-                if (_isEnumerable && this.method.Parameters.Any(static p => p.IsSourceParameterWithEnumeratorCancellationAttribute()))
+                if (_isEnumerable && this.method.Parameters.Any(static p => !p.IsExtensionParameterImplementation() && p.HasEnumeratorCancellationAttribute))
```

Comment:
>!p.IsExtensionParameterImplementation() [](http://example.com/codeflow?start=76&length=39)

It is not obvious why extension parameter is getting special treatment

Reply:
Do you mean we should leave a comment here (the attribute is ignored on extension parameter) or do you mean you think the attribute should be effective on the extension parameter too?

The reason I don't think the attribute should be effective is that this attribute is very specific to async-iterator method implementations. I don't think it belongs on an extension parameter, as the extension block may contain many kinds of members. Also, the scenario where you need an async-iterator extension method on a `CancellationToken` receiver seems very niche to say the least.

Reply:
> I don't think it belongs on an extension parameter

I am fine with this design decision, Could you please mention this explicitly in the speclet?

Also, please make sure to test scenario where extension parameter implementation is the only parameter for the method.

Reply:
Sounds good. Linked to corresponding spec update.
The test `ReceiverParameterValidation_CancellationTokenParameter_Instance` shows that the attribute doesn't have an effect (it includes another parameter, but I think that's fine)

Reply:
> it includes another parameter, but I think that's fine

Are you saying the `CancellationToken` parameter doesn't have to be the last parameter in a general case?

Reply:
That's correct. The only restriction is that only one parameter may be marked with this attribute.